Driver post ride

whyderrick opened the pull request.

whyderrick committed:
Implement ride creation

This commit adds the ability to create new rides as a user. Here's the
changes:
- It defines routing resources in routes.rb
- It creates controllers for the ride and user models. It also
establishes `#new`, `#show`, and `#index` actions in the ride controller
- It creates basic views for those actions.
- Lastly, it adds the HAML-Rails gem for view templating.

whyderrick committed:
Minor tweak to rides index view.

bootcoder commented on app/controllers/rides_controller.rb:
```diff
@@ -0,0 +1,52 @@
+class RidesController < ApplicationController
+  def index
+    @rides = Ride.all
+    respond_to do |format|
```
Pretty sure the index works without the respond to block. Try it....

bootcoder commented on app/controllers/rides_controller.rb:
```diff
@@ -0,0 +1,52 @@
+  end
+
+  def show
+    @ride = Ride.find(params[:id])
```
DRY this out with a before_action ala
```ruby 
  before_action :set_user, only: [:show, :edit, :update, :destroy]
```
followed by
```ruby 
  def set_user
    @user = User.find(params[:id])
  end
```

bootcoder commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,22 @@
+class UsersController < ApplicationController
```
You're not doing anything with this controller nor is it your responsibility.......
rm -rf

bootcoder commented on app/views/rides/index.html.haml:
```diff
@@ -0,0 +1,5 @@
+%h1 Rides index
+
+- @rides.each do |item|
```
IMHO ul-li is the better tag here

bootcoder reviewed (commented):
Couple things, mostly awesome. ☯️

whyderrick committed:
Resolve Code Review comments

This commit makes changes in line with most of Hunter's comments on the
repo's PR #1 Specifically, it adds a before_action filter and a private
`#set_user` method and remove manual actions of setting the user. It
changes the rides index view to swap <ul> tags for <p> tags. It deletes
this branch's empty UserController file.

The one unaddressed comment was regarding Rails default behavior with
request format specification and it needs the respond_to block to work.

whyderrick commented:
Merging in after addressing PR comments.
